solver/src/http_solver.rs

```diff
@@ -130,36 +130,43 @@ impl HttpSolver {
             default_fee: 0.0,
         })
     }
-}
 
-#[async_trait::async_trait]
-impl Solver for HttpSolver {
-    async fn solve(&self, orders: Vec<Liquidity>) -> Result<Option<Settlement>> {
+    async fn send(&self, model: &BatchAuctionModel) -> Result<SettledBatchAuctionModel> {
         let mut url = self.base.clone();
         url.set_path("/solve");
         self.config.add_to_query(&mut url);
-        let body = self.create_body(orders.as_slice()).await?;
+        let query = url.query().map(ToString::to_string).unwrap_or_default();
         let mut request = self.client.post(url);
         if let Some(api_key) = &self.api_key {
             request = request.header("X-API-KEY", api_key);
         }
-        let request = request.json(&body);
+        let body = serde_json::to_string(&model).context("failed to encode body")?;
+        tracing::debug!("request query {} body {}", query, body);
+        let request = request.body(body);
 
-        let response = request
-            .send()
-            .await
-            .context("failed to send solver request")?
-            .error_for_status()
-            .context("solver response was unsuccessful")?;
-        let body = response
+        let response = request.send().await.context("failed to send request")?;
+        let status = response.status();
+        let body_bytes = response
             .bytes()
             .await
             .context("failed to get response body")?;
-        let _decoded: SettledBatchAuctionModel =
-            serde_json::from_slice(&body).with_context(|| match std::str::from_utf8(&body) {
-                Ok(body) => format!("failed to decode response body: {}", body),
-                Err(_) => format!("failed to decode response body: {:?}", body),
-            })?;
+        let body_str =
```

Comment:
why not use `respone.text()` or even `response.json()` directly? (the latter I assume because you want to print the intermediate representation for debugging)

Reply:
fixed.